### PR TITLE
Added recusion to handler copy + debugPrint fn

### PR DESCRIPTION
**Pull request by rgee0**
Signed-off-by: rgee0 <richard@technologee.co.uk>

Addresses issue #59.  

## Description
Changes the recursive arg to true on the handler `copyFiles` call so that sub directories within the handler are also copied into the build directory. 
Extracted a `debugPrint` function to avoid code repetition as The new directory branch in `copyFiles` now messages out only if the system is creating a new directory and `debug` is enabled.
Corrected a typo which meant that `err` was being tested rather than `newDirErr` when a new directory is being created.
Changed FaaS to OpenFaaS where seen.

## Motivation and Context
#59
- [X] There is a raised an issue to propose this change ([required](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Created 10 levels of files and folders beneath a handler folder
Ran ./faas-cli -action build -yaml ./samples.yml
Confirmed that the images built successfully.
Confirmed that the expected files and folders were extant in the build folder following completion of build

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks for the fixes. If all usages are recursive then the flag is redundant and can be removed.

**Comment by alexellis** [comment]
Can we get any test coverage for this scenario?

File: builder/build.go [hunk 1]
```diff
@@ -161,3 +160,10 @@ func buildFlagString(nocache bool, squash bool, httpProxy string, httpsProxy str
 
 	return buildFlags
 }
+
+func debugPrint(message string) {
+
+	if val, exists := os.LookupEnv("debug"); exists && (val == "1" || val == "true") {
```
Comment (johnmccabe) [inline builder/build.go]: Could add a `--debug` flag for turning this on across all commands, something for a seperate PR perhaps.

**Comment by johnmccabe** [comment]
have built and run a few template builds locally and it looks good 👍

File: builder/build.go (hunk 1 above)
Comment (rgee0) [reply builder/build.go]: Yes, that was kind of the theory behind pulling it out.  Suggest a separate PR since this one is a 'bug' fix

**Comment by alexellis** [comment]
Thanks for helping to integration test. I would like some automated tests in "Golang" for this.. doesn't have to be super extensive. Can either of you take that on?

**Pull request merged**